proj-android/PowerAuthLibrary/src/main/java/io/getlime/security/powerauth/sdk/PowerAuthSDK.java

```diff
@@ -166,13 +167,14 @@ public PowerAuthSDK build(@NonNull final Context context) {
                 instance.mClientConfiguration = new PowerAuthClientConfiguration.Builder().build();
             }
             instance.mClient = new HttpClient(instance.mClientConfiguration, instance.mConfiguration.getBaseEndpointUrl(), new DefaultExecutorProvider());
-            instance.mStatusKeychain = new PA2Keychain(instance.mKeychainConfiguration.getKeychainStatusId());
-            instance.mBiometryKeychain = new PA2Keychain(instance.mKeychainConfiguration.getKeychainBiometryId());
+            instance.mStatusKeychain = KeychainFactory.getKeychain(appContext, instance.mKeychainConfiguration.getKeychainStatusId());
+            instance.mBiometryKeychain = KeychainFactory.getKeychain(appContext, instance.mKeychainConfiguration.getKeychainBiometryId());
+            instance.mTokenStoreKeychain = KeychainFactory.getKeychain(appContext, instance.mKeychainConfiguration.getKeychainTokenStoreId());
 
             if (mStateListener != null) {
                 instance.mStateListener = mStateListener;
             } else {
-                instance.mStateListener = new DefaultSavePowerAuthStateListener(context, instance.mStatusKeychain);
+                instance.mStateListener = new DefaultSavePowerAuthStateListener(instance.mStatusKeychain);
```

Comment:
I would check this with @TomasKypta - I think that in the last discussion, Tom misunderstood the question and by "storing context", he understood "persistently storing context" (as opposed to "passing context as a parameter, vs. storing in a volatile local variable"). Many Android SDK APIs pass context as the method call parameter...

Reply:
Well, there's a difference between context and an application context. Keeping the application context (that's what we do here), is OK. We did this also in the previous implementation. It's also not recommended to pass application context to every call.

Reply:
Previously `DefaultSavePowerAuthStateListener` kept persistently application context, now it's moved to the `Keychain` implementation. The result is the same, we still keep the context relevant for the `PowerAuthSDK` instance.